src/strands/event_loop/_recover_message_on_max_tokens_reached.py

```diff
@@ -14,39 +14,44 @@
 
 
 def recover_message_on_max_tokens_reached(message: Message) -> Message:
-    """Recover and clean up incomplete messages when max token limits are reached.
+    """Recover and clean up messages when max token limits are reached.
 
-    When a model response is truncated due to maximum token limits, tool use blocks may be
-    incomplete or malformed. This function inspects the message content and:
+    When a model response is truncated due to maximum token limits, all tool use blocks
+    should be replaced with informative error messages since they may be incomplete or
+    unreliable. This function inspects the message content and:
 
-    1. Identifies incomplete tool use blocks (missing name, input, or toolUseId)
-    2. Replaces incomplete tool uses with informative error messages
-    3. Preserves all valid content blocks (text and complete tool uses)
+    1. Identifies all tool use blocks (regardless of validity)
+    2. Replaces all tool uses with informative error messages
+    3. Preserves all non-tool content blocks (text, images, etc.)
     4. Returns a cleaned message suitable for conversation history
 
     This recovery mechanism ensures that the conversation can continue gracefully even when
-    model responses are truncated, providing clear feedback about what happened.
+    model responses are truncated, providing clear feedback about what happened and preventing
+    potentially incomplete or corrupted tool executions.
+
+    TODO: after https://github.com/strands-agents/sdk-python/issues/561 is completed, only the verifiable
```

Comment:
I think we wouldn't want to change this behavior FYI - it would be a breaking change as it would force folks to *need* to resume instead of re-prompting after.

I think this behavior is going to be permanent and if they want different behavior, we need to come up with a pluggable solution